nomos-services/blend/src/core/mod.rs

```diff
@@ -649,9 +639,32 @@ where
         saved_state
     } else {
         tracing::debug!(target: LOG_TARGET, "No recovery state found for session {:?}. Initializing a new one.", current_membership_info.public.session);
-        ServiceState::with_session(current_membership_info.public.session, state_updater)
+
+        ServiceState::with_session(
+            current_membership_info.public.session,
+            SessionBlendingTokenCollector::new(
+                &reward::SessionInfo::new(
+                    current_membership_info.public.session,
+                    &pol_epoch_nonce,
+                    current_membership_info.public.membership.size() as u64,
+                    current_membership_info.public.poq_core_public_inputs.quota,
+                ).expect("Reward session info must be created successfully. Panicking since the service cannot continue with this session")
+            ),
+            None,
+            state_updater,
+        ).expect("service state should be created successfully")
     };
 
+    // If there is the old session token collector loaded from `last_saved_state`,
+    // compute/submit its activity proof because we won't collect more tokens for
+    // the old session after this initialization step.
```

Comment:
Can we add to the comment that we won't collect because we are not establishing connections with peers from the old session?

Reply:
Resolved c00a9a14b2d30650e6c53916f44fcef706d1b90a